src/ng/compile.js

```diff
@@ -785,6 +786,9 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     var bindings = {};
 
     forEach(scope, function(definition, scopeName) {
+      if (scopeName in bindingCache) {
```

Comment:
Since `bindingCache` is global (as in "$compiler-wide"), what if different isolate bindings, have different `definition` for the same `scopeName` ?
Am I missing something ?